dali/operators/segmentation/random_mask_pixel.cc

```diff
@@ -0,0 +1,203 @@
+                              uint64_t, int64_t, float)
+
+namespace dali {
+
+DALI_SCHEMA(segmentation__RandomMaskPixel)
+    .DocStr(R"(Selects random pixel coordinates in a mask, sampled from a uniform distribution.
+
+Based on run-time argument ``foreground``, it returns either only foreground pixels or any pixels.
+
+Pixels are classificed as foreground either when their value exceeds a given ``threshold`` or when
+it's equal to a specific ``value``.
+)")
+    .AddOptionalArg<int>("value",
+      R"code(All pixels equal to this value are  interpreted as foreground.
+
+This argument is mutually exclusive with ``threshold`` argument and is meant to be used only
+with integer inputs.
+)code", nullptr, true)
+    .AddOptionalArg<float>("threshold",
+      R"code(All pixels with a value above this threshold are interpreted as foreground.
+
+This argument is mutually exclusive with ``value`` argument.
+)code", 0.0f, true)
+    .AddOptionalArg("foreground",
+      R"code(If different than 0, the pixel position is sampled uniformly from all foreground pixels.
+If 0, the pixel position is sampled uniformly from all available pixels.)code",
+      0, true)
+    .NumInput(1)
+    .NumOutput(1);
+
+class RandomMaskPixelCPU : public Operator<CPUBackend> {
+ public:
+  explicit RandomMaskPixelCPU(const OpSpec &spec);
+  bool CanInferOutputs() const override { return true; }
+  bool SetupImpl(std::vector<OutputDesc> &output_desc, const workspace_t<CPUBackend> &ws) override;
+  void RunImpl(workspace_t<CPUBackend> &ws) override;
+
+ private:
+  template <typename T>
+  void RunImplTyped(workspace_t<CPUBackend> &ws);
+
+  int64_t seed_;
+  std::vector<std::mt19937_64> rng_;
+  std::vector<SearchableRLEMask> rle_;
+
+  std::vector<int> foreground_;
+  std::vector<int> value_;
+  std::vector<float> threshold_;
+
+  bool has_value_ = false;
+
+  USE_OPERATOR_MEMBERS();
+};
+
+RandomMaskPixelCPU::RandomMaskPixelCPU(const OpSpec &spec)
+    : Operator<CPUBackend>(spec),
+      seed_(spec.GetArgument<int64_t>("seed")),
+      has_value_(spec.ArgumentDefined("value")) {
+  if (has_value_) {
+    DALI_ENFORCE(!spec.ArgumentDefined("threshold"),
+      "Arguments ``value`` and ``threshold`` can not be provided together");
+  }
+}
+
+bool RandomMaskPixelCPU::SetupImpl(std::vector<OutputDesc> &output_desc,
+                                    const workspace_t<CPUBackend> &ws) {
+  const auto &in_masks = ws.template InputRef<CPUBackend>(0);
+  int nsamples = in_masks.size();
+  auto in_masks_shape = in_masks.shape();
+  int ndim = in_masks_shape.sample_dim();
+  output_desc.resize(1);
+  output_desc[0].shape = uniform_list_shape(nsamples, {ndim});
+  output_desc[0].type = TypeTable::GetTypeInfo(DALI_INT64);
+
+  foreground_.resize(nsamples);
+  value_.clear();
+  threshold_.clear();
+
+  for (int sample_idx = 0; sample_idx < nsamples; sample_idx++) {
+    foreground_[sample_idx] = spec_.template GetArgument<int>("foreground", &ws, sample_idx);
+  }
```

Comment:
Sample-indexed `GetArgument` is a relic from the obsolete SampleWorkspace. It's inefficient and should not be used.
```suggestion
  GetPerSampleArgument(foreground_, "foreground", spec_, ws, nsamples);
```